parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java

```diff
@@ -4593,15 +4605,84 @@ private void checkExNotNull(SqlParserUtil.StringAndPos sap, Throwable thrown) {
     sql(sql).fails(expected);
   }
 
-  @Test public void testAllocateCursor() {
-    final String sql = "create procedure foo ()\n"
-        + "begin\n"
-        + "allocate bar cursor for procedure baz;\n"
-        + "end";
-    final String expected = "CREATE PROCEDURE `FOO` ()\n"
-        + "BEGIN\n"
-        + "ALLOCATE `BAR` CURSOR FOR PROCEDURE `BAZ`;\n"
-        + "END";
+  @Test public void testCaseStmtWithOperandBase() {
+    final String sql = "create procedure foo (bee integer) "
+        + "case bee "
+        + "when 2 then "
+        + "select bar; "
+        + "when 3 then "
+        + "select abc; "
+        + "end case";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER)\n"
+            + "CASE `BEE` WHEN 2 THEN SELECT `BAR`;\n"
+            + "WHEN 3 THEN SELECT `ABC`;\n"
+            + "END CASE";
     sql(sql).ok(expected);
   }
+
+  @Test public void testCaseStmtWithOperandWithElse() {
+    final String sql = "create procedure foo (bee integer) "
+        + "case bee "
+        + "when 2 then "
+        + "select bar; "
+        + "when 3 then "
+        + "select abc; "
+        + "else "
+        + "select cde;"
+        + "end case";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BEE` INTEGER)\n"
+            + "CASE `BEE` WHEN 2 THEN SELECT `BAR`;\n"
+            + "WHEN 3 THEN SELECT `ABC`;\n"
+            + "ELSE SELECT `CDE`;\n"
+            + "END CASE";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testCaseStmtWithConditionalExpressionBase() {
```

Comment:
Can you nest case statements?

Reply:
Yes, case statements can be nested. I've added a test case at https://github.com/googleinterns/calcite/pull/263/commits/3d8e35213af09f1537fd4d58f187acc458f4c37f parsing a nested case statement.